### PR TITLE
Upgrade deps, update eslint config

### PR DESCRIPTION
**Pull request by runk**
#### What's this PR do?

Explain **WHAT** you are doing in this PR and **WHY**.

Any background context you want to provide?

#### References

Provide all necessary information you think reviewer would want. Relevant tickets?

 - Jira/Trello/GH Issue reference #
 - Related PR #

#### Testing scope

If PR should be tested manually, explain what should be done.

 - [ ] Check A
 - [ ] Check B
 - [ ] Check C

#### Screenshots 

Screenshots if any?

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,11 +4,29 @@
...
   "rules": {
-    "id-length": [2, {"min": 2, "properties": "never", "exceptions": ["_", "i", "j"]}],
-    "indent": [2, 2, {"SwitchCase": 1}],
-    "max-len": [2, 120],
-    "no-shadow": [2, { "allow": ["cb", "err", "done", "resolve", "reject"] }],
-    "import/no-extraneous-dependencies": ["error", {devDependencies: true}],
+    "comma-dangle": ["error", {
+      "arrays": "always-multiline",
+      "objects": "always-multiline",
+      "functions": "never", # Node 6 doesn't like `(function(a,){ })(b,);`
```
Comment (camjc) [inline .eslintrc]: I still don't understand this. Why can't you make it `always-multiline` too?

File: .eslintrc [hunk 2]
```diff
@@ -4,11 +4,29 @@
...
+    "jsdoc/check-tag-names": "error",
+    "jsdoc/check-types": "error",
+    "jsdoc/newline-after-description": "error",
+    "jsdoc/require-description-complete-sentence": "error",
+    "jsdoc/require-hyphen-before-param-description": "error",
+    "jsdoc/require-param": "error",
+    "jsdoc/require-param-description": "error",
+    "jsdoc/require-param-type": "error",
+    "jsdoc/require-returns-description": "error",
+    "jsdoc/require-returns-type": "error",
```
Comment (camjc) [inline .eslintrc]: Awesome you've added jsdoc. would be cool to do lodash too

File: .eslintrc (hunk 1 above)
Comment (runk) [inline .eslintrc]: There's a comment

File: .eslintrc (hunk 2 above)
Comment (runk) [inline .eslintrc]: Lodash isn't a dependency at the momeent

Comment (camjc) [inline .eslintrc]: Fair enough. Cool mate.

**Pull request merged**